---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Execute entire block - *Cmd+Shift+Enter*. \

Add a new chunk by pressing *Cmd+Option+I*.

Packages
```{r}
library(tidyverse)
library(ggplot2)
library(plotly)
library(ggpubr)
library(Seurat)
```
Reads in dataset
```{r}
human <- read_csv('../data/20230602-human_data.csv')
nhp <- read_csv('../data/20230602-nhp_data.csv')
mouse <- read_csv('../data/20230602-mouse_data.csv')
orthologs <- read_csv('../data/20230607-orthologs-hm_nhp_ms.csv')
```
Human bar plot
```{r}
colors <- c('gray', 'purple', 'violet', 'orange3', 'pink3', 'olivedrab4', 'plum4', 'blue', 'green', 'red')

level_order <- c('Fibroblasts', 'Acinar', 'Ductal', 'Stellate', 'Immune', 'Endothelial', 'Gamma', 'Epsilon', 'Delta', 'Beta', 'Alpha')

human$celltype <- factor(human$celltype, levels = level_order)

human %>% 
  filter(Gene == 'INS') %>% 
  ggplot(aes(x=celltype, y = avg_expr, fill = celltype)) + 
  geom_col(position = position_dodge()) +
  coord_flip() +
  theme_minimal() +
  theme(axis.title.y = element_blank(),
        legend.position = 'none') +
  labs(y = 'Scaled Expression') +
  fill_palette(palette = colors) +
  scale_x_discrete(drop = F)

```

creating ortholog dictionaries
```{r}
human_genes <- human %>% 
  select(Gene) %>% 
  distinct() %>% 
  left_join(orthologs, by = c('Gene' = 'Gene name'))
write_csv(human_genes, '../data/gene_dict_df.csv')
nhp_orthos <- with(human_genes, setNames(`Macaque gene name`, `Gene`))
ms_orthos <- with(human_genes, setNames(`Mouse gene name`, `Gene`))
```
Non-human Primate bar plot
```{r}
nhp$celltype <- factor(nhp$celltype, levels = level_order)

nhp %>% 
  filter(Gene == nhp_orthos['INS']) %>% 
  ggplot(aes(x=celltype, y = avg_expr, fill = celltype)) + 
  geom_col() +
  coord_flip() +
  theme_minimal() +
  theme(axis.title.y = element_blank(),
        legend.position = 'none') +
  labs(y = 'Scaled Expression') +
  fill_palette(palette = colors) +
  scale_x_discrete(drop = F)
```
Mouse bar plot
```{r}
mouse$celltype <- factor(mouse$celltype, levels = level_order)

mouse %>% 
  filter(Gene == ms_orthos['INS']) %>% 
  ggplot(aes(x=celltype, y = avg_expr, fill = celltype)) + 
  geom_col() +
  coord_flip() +
  theme_minimal() +
  theme(axis.title.y = element_blank(),
        legend.position = 'none') +
  labs(y = 'Scaled Expression') +
  fill_palette(palette = colors) +
  scale_x_discrete(drop = F)
```

```{r}
human_longer <- human %>% 
  mutate(species = 'human') %>% 
  pivot_longer(cols = pct_expr:scaled_expr, names_to = 'scale_types', values_to = 'expr') %>% 
  left_join(orthologs, by = c('Gene' = 'Gene name')) 

nhp_longer <- nhp %>% 
  mutate(species = 'nhp') %>% 
  pivot_longer(cols = pct_expr:scaled_expr, names_to = 'scale_types', values_to = 'expr')

mouse_longer <- mouse %>% 
  select(Gene, celltype, pct_expr, avg_expr, scaled_expr) %>% 
  mutate(species = 'mouse') %>% 
  pivot_longer(cols = pct_expr:scaled_expr, names_to = 'scale_types', values_to = 'expr')

all <- bind_rows(human_longer, nhp_longer, mouse_longer)

# all <- all_comb %>% 
#   left_join(orthologs, by = c('Gene' = 'Gene name'))
```

```{r}
colors <- c('brown', 'gray', 'purple', 'violet', 'orange3', 'pink3', 'olivedrab4', 'plum4', 'blue', 'green', 'red')

level_order <- c('Fibroblasts', 'Acinar', 'Ductal', 'Stellate', 'Immune', 'Endothelial', 'Gamma', 'Epsilon', 'Delta', 'Beta', 'Alpha')

all$celltype <- factor(all$celltype, levels = level_order)

all %>% 
  filter(scale_types == 'pct_expr',
         Gene == 'INS') %>% 
  ggplot(aes(x = celltype, y = expr, fill = celltype)) +
  geom_col() +
  coord_flip() + 
  facet_wrap(vars(species), ncol = 3) +
  theme_minimal() +
  theme(axis.title.y = element_blank(),
        legend.position = 'none') +
  labs(y = 'Scaled Expression') +
  fill_palette(palette = colors) +
  scale_x_discrete(drop = F)

```

filter metadata for info charts
```{r}
metadata <- read_csv('../data/species_metadata.csv')

test <- metadata %>% 
  select(Species, `GEO ID`, Citation, Platform, Donors, total_cells) %>% 
  distinct() %>% 
  t() 
# test <- rest %>% 
#     mutate(`Dataset label` = rownames(test))


```

metadata plot - pie chart
```{r}
metadata %>% 
  filter(Species == 'Mus musculus') %>% 
  ggplot(aes(x = "", y = total, fill = celltype)) +
  geom_bar(stat = 'identity', width = 1, color = 'white') +
  coord_polar('y', start = 0) +
  theme_void() #+
# theme(axis.text = element_blank(),
#       axis.ticks = element_blank(),
#       panel.grid = element_blank(),
#       axis.title = element_blank())
```

metadata donut chart
```{r}
# metadataDonut <- 
metadata %>% 
  filter(Species == 'Homo sapiens') %>% 
  mutate(ymax = cumsum((perc_total/100))) %>% 
  mutate(ymin = c(0, head(ymax, n=-1))) %>% 
  ggplot(aes(ymax = ymax, ymin=ymin, xmax = 4, xmin = 3, fill = celltype)) +
  geom_rect(color = 'white') +
  coord_polar('y') +
  xlim(c(1.5,4)) +
  theme_void() +
  labs(fill = 'Cell Type') +
  theme(legend.position = 'none',
        plot.margin = unit(c(-0.5, -0.5, -0.5, -0.5), 'inches'))


```

```{r}
test <- data.frame(
  category=c("A", "B", "C"),
  count=c(10, 60, 30)
)

test$yfraction = test$count/sum(test$count)

test$ymax = cumsum(test$yfraction)
test$ymin = c(0, head(test$ymax, n = -1))

```

